creator-node/src/services/stateMachineManager/monitoring/StateMonitoringQueue.js

```diff
@@ -0,0 +1,254 @@
+const BullQueue = require('bull')
+const _ = require('lodash')
+
+const config = require('../../../config')
+const {
+  QUEUE_HISTORY,
+  STATE_MONITORING_QUEUE_NAME,
+  STATE_MONITORING_QUEUE_MAX_JOB_RUNTIME_MS,
+  STATE_MONITORING_QUEUE_INIT_DELAY_MS
+} = require('../constants')
+const { logger } = require('../../../logging')
+const redis = require('../../../redis')
+const { getLatestUserId } = require('./utils')
+const processStateMonitoringJob = require('./processStateMonitoringJob')
+
+/**
+ * Handles setup and lifecycle management (producing/adding jobs and consuming/ processing jobs)
```

Comment:
we should be intentional about adding new terminology here, overloading adding/processing with producing/consuming - i'm fine with it but we should then use this naming everywhere, and have a wrapper function that just calls .add or .process and not expose that name anywhere else

is producer/consumer an industry standard? job consumer is a bit confusing conceptually, maybe producer / processor, but up to you

Reply:
synced offline -- changing to add/process and remove producer/consumer to match Bull's terminology